### PR TITLE
Updated existing keywords with condition for dynamic multi store

### PR DESCRIPTION
**Pull request by Nataliia-Popkova**
Updated existing keywords with condition for dynamic multi store

## Checklist
- [ x ] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: resources/common/common_yves.robot [hunk 1]
```diff
@@ -182,9 +182,22 @@ Yves: go to the 'Home' page
...
     Set Browser Timeout    ${browser_timeout}
-    Go To    ${yves_at_url}
+    ${dms_state}=    Convert To String    ${dms}
+    IF    '${dms_state}' != 'True'
+        Go To    ${yves_at_url}
+    ELSE
+        Go To    ${yves_url}
+        Wait Until Element Is Visible    ${store_switcher_header_menu_item}
+        IF    '${env}' in ['ui_b2b','ui_mp_b2b']
+            Click Element with JavaScript:    ${store_switcher_header_menu_item}
```
Comment (AntonShopin) [inline resources/common/common_yves.robot:195]: Do we need to click select with JS if the next step is `Select From List By Value`, which doesn't need select to be expanded?

File: resources/common/common_yves.robot [hunk 2]
```diff
@@ -182,9 +182,22 @@ Yves: go to the 'Home' page
...
+        ELSE
+            Click    ${store_switcher_header_menu_item}
```
Comment (AntonShopin) [inline resources/common/common_yves.robot:197]: same here - why do we need to click select first?

File: resources/common/common_yves.robot [hunk 3]
```diff
@@ -182,9 +182,22 @@ Yves: go to the 'Home' page
...
+        END
+        Select From List By Value    ${store_switcher_header_menu_item}    ${store}
```
Comment (AntonShopin) [inline resources/common/common_yves.robot:199]: Can you confirm, that once store is selected in the select, page is reloaded and Robot automatically waits for the page reload?

File: resources/common/common_yves.robot [hunk 4]
```diff
@@ -205,11 +218,25 @@ Yves: go to URL:
...
+    ${dms_state}=    Convert To String    ${dms}
+    IF   '${dms_state}' != 'True'
+        Go To    ${yves_at_url}${url}
+    ELSE
+        Go To    ${yves_url}
+        Wait Until Element Is Visible    ${store_switcher_header_menu_item}
+            IF    '${env}' in ['ui_b2b','ui_mp_b2b']
+                Click Element with JavaScript:    ${store_switcher_header_menu_item}
+            ELSE
+                Click    ${store_switcher_header_menu_item}
```
Comment (AntonShopin) [inline resources/common/common_yves.robot:234]: same comments as for `Yves: go to AT store 'Home' page if other store not specified:`

File: resources/common/common_yves.robot [hunk 5]
```diff
@@ -228,10 +255,25 @@ Yves: go to newly created page by URL:
         END
     END
 
-Yves: go to newly created page by URL on AT store:
-    [Arguments]    ${url}    ${delay}=5s    ${iterations}=31
+Yves: go to newly created page by URL on AT store if other store not specified:
+    [Arguments]    ${url}    ${delay}=5s    ${iterations}=31    ${store}=AT
     FOR    ${index}    IN RANGE    1    ${iterations}
-        Go To    ${yves_at_url}${url}?${index}
+        ${dms_state}=    Convert To String    ${dms}
```
Comment (AntonShopin) [inline resources/common/common_yves.robot:261]: This line should be before the loop

File: resources/common/common_yves.robot [hunk 6]
```diff
@@ -228,10 +255,25 @@ Yves: go to newly created page by URL:
...
+        IF   '${dms_state}' != 'True'
+            Set Browser Timeout    ${browser_timeout}
```
Comment (AntonShopin) [inline resources/common/common_yves.robot:263]: This keyword should be before the loop and not multiple times inside

File: resources/common/common_yves.robot [hunk 7]
```diff
@@ -228,10 +255,25 @@ Yves: go to newly created page by URL:
...
+            Go To    ${yves_at_url}${url}?${index}
+        ELSE
+            Set Browser Timeout    ${browser_timeout}
+            Go To    ${yves_url}
+            Wait Until Element Is Visible    ${store_switcher_header_menu_item}
+                IF    '${env}' in ['ui_b2b','ui_mp_b2b']
+                    Click Element with JavaScript:    ${store_switcher_header_menu_item}
+                ELSE
+                    Click    ${store_switcher_header_menu_item}
```
Comment (AntonShopin) [inline resources/common/common_yves.robot:272]: same as before. Do we need to click select to select its option

File: resources/common/common_yves.robot [hunk 8]
```diff
@@ -245,10 +287,25 @@ Yves: go to newly created page by URL on AT store:
         END
     END
 
-Yves: go to URL and refresh until 404 occurs:
-    [Arguments]    ${url}    ${delay}=5s    ${iterations}=31
+Yves: go to store specified URL and refresh until 404 occurs:
```
Comment (AntonShopin) [inline resources/common/common_yves.robot:290]: same comments as for `Yves: go to newly created page by URL on AT store if other store not specified:`

File: resources/pages/yves/yves_header_section.robot [hunk 9]
```diff
@@ -4,6 +4,7 @@ ${user_navigation_menu_login_button}    xpath=//a[@class='user-block__button but
...
 &{user_navigation_fly_out_header_menu_item}    ui_b2b=xpath=//li[contains(@class,'user-navigation__item--user')]//nav[contains(@class,'user-navigation__sub-nav')]//ul[contains(@class,'list--secondary')]    ui_b2c=xpath=//div[contains(@class,'user-block js-nav-overlay__drop-down-block')]    ui_mp_b2b=xpath=//li[contains(@class,'user-navigation__item--user')]//nav[contains(@class,'user-navigation__sub-nav')]//ul[contains(@class,'list--secondary')]    ui_mp_b2c=xpath=//div[contains(@class,'user-block js-nav-overlay__drop-down-block')]
 ${company_name_icon_header_menu_item}    xpath=//div[@class='header__top']//a[contains(@class,'navigation-top__company')]
+${store_switcher_header_menu_item}    xpath=//*[@data-qa='component header']//select[@name='_store']
```
Comment (AntonShopin) [inline resources/pages/yves/yves_header_section.robot:7]: please change the selector on smth like `xpath=//*[@data-qa='component header']//select[conains(@name,'store')]`

File: resources/steps/header_steps.robot [hunk 10]
```diff
@@ -20,6 +20,7 @@ ${shoppingCartIcon}    ${shopping_car_icon_header_menu_item}[${env}]
 ${customerSearchWidget}    ${agent_customer_search_widget}
 ${quoteRequestsWidget}    ${agent_quote_requests_header_item}
 ${wishlistIcon}    ${wishlist_icon_header_navigation_widget}
+${storeSwitcher}    ${store_switcher_header_menu_item}
```
Comment (AntonShopin) [inline resources/steps/header_steps.robot:23]: what it's for?

File: tests/ui/e2e/b2b.robot [hunk 11]
```diff
@@ -1407,35 +1406,33 @@ Multistore_Product
     ...    || productAbstract   | unselect store ||
     ...    || multiSKU${random} | AT             ||
     Trigger multistore p&s
-    Yves: go to URL and refresh until 404 occurs:    ${url}
+    Yves: go to store specified URL and refresh until 404 occurs:    ${url}
```
Comment (AntonShopin) [inline tests/ui/e2e/b2b.robot:1409]: `Yves: go to store specified URL and refresh until 404 occurs:` 
if the keyword is called like this, then we need to pass store from tests, no?

File: tests/ui/e2e/suite.robot [hunk 12]
```diff
@@ -51,3677 +51,3667 @@ Resource    ../../../resources/steps/merchants_steps.robot
...
+# Guest_User_Access_Restrictions
+#     [Documentation]    Checks that guest users see products info and cart but not profile
+#     Yves: header contains/doesn't contain:    true    ${currencySwitcher}[${env}]    ${shoppingCartIcon}
+#     Yves: go to PDP of the product with sku:    002
+#     Yves: PDP contains/doesn't contain:     true    ${pdpPriceLocator}    ${addToCartButton}
+#     Yves: add product to the shopping cart
+#     Yves: go to b2c shopping cart
+#     Yves: shopping cart contains product with unit price:   sku=002    productPrice=99.99    productName=Canon IXUS 160
+#     Yves: go to user menu:    Overview
+#     Yves: 'Login' page is displayed
```
Comment (AntonShopin) [inline tests/ui/e2e/suite.robot]: all tests for suite a commented out

**Review by AntonShopin — CHANGES_REQUESTED** [review]

File: resources/common/common_yves.robot (hunk 3 above)
Comment (Nataliia-Popkova) [reply resources/common/common_yves.robot:194]: Added: Wait Until Element Contains    //*[@data-qa='component header']//select[contains(@name,'store')]/option[@selected='']    ${store}

**Review by AntonShopin — APPROVED** [review]

**Pull request merged**